src/platform.ts

```diff
@@ -162,10 +189,30 @@ export function getAvailablePowerShellExes(
         }
     } else {
         // Handle Linux and macOS case
+        const defaultExePath =  this.getDefaultPowerShellPath(platformDetails);
         paths.push({
-            versionName: "PowerShell Core",
-            exePath: this.getDefaultPowerShellPath(platformDetails),
+            versionName: "PowerShell Core" + (/-preview/.test(defaultExePath) ? " Preview" : ""),
+            exePath: defaultExePath,
         });
+
+        // If defaultExePath is pwsh, check to see if pwsh-preview is also installed and if so, make it available.
+        // If the defaultExePath is already pwsh-preview, then pwsh is not installed - nothing to do.
+        let osExePath;
+        let osPreviewExePath;
+        if (platformDetails.operatingSystem === OperatingSystem.MacOS) {
+            osExePath = macOSExePath;
+            osPreviewExePath = macOSPreviewExePath;
+        } else if (platformDetails.operatingSystem === OperatingSystem.Linux) {
+            osExePath = linuxExePath;
```

Comment:
do we not need to do the same ternary op here to check for snap?

Reply:
Not needed here because `getDefaultPowerShellPath` already returns just one of: linuxExePath, linuxPreviewExePath, snapExePath, snapPreviewExePath.  It searches in that order.

Reply:
BTW I get the impression (perhaps incorrect impression) that on a Snap system, PowerShell will only be at /snap/bin/pwsh* and that you wouldn't also have PowerShell at /usr/bin/pwsh*.  And on a straight (non-Snap) Linux system, there would be no /snap/bin/pwsh*.  That is, these two sets are mutually exclusive. Is that correct?

Reply:
So snap is really just a package manager that you first install with apt. So you _could_ have installed `pwsh` by Snap and Apt on the same system.

Not quite sure why one would do that... but it's physically possible. I guess I could see something along the lines of installing pwsh with apt and pwsh-preview with snap maybe...

So it's physically possible to have a `/snap/bin/pwsh` and a `/usr/bin/pwsh`.